packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -149,12 +149,7 @@ function h(sel: string, data: VElementData, children: VNodes = EmptyArray): VEle
             data.styleDecls && data.style,
             `vnode.data.styleDecls and vnode.data.style ambiguous declaration.`
         );
-        if (data.style && !isString(data.style)) {
-            logError(
-                `Invalid 'style' attribute passed to <${sel}> is ignored. This attribute must be a string value.`,
-                vmBeingRendered
-            );
-        }
+
```

Comment:
I ended up moving this check into the `patchStyleAttribute` routine instead because there isn't an easy way to retrieve the tag name inside the `sp` function.

I've created a test to ensure it provides the same results.